File: JavaScript/2dArrayQuestions/hourglassSum.js
function hourglassSum(matrix) {
    const rows = matrix.length;
    const cols = matrix[0].length;
    let maxSum = -Infinity;
    for (let i = 0; i < rows - 2; i++) {
        for (let j = 0; j < cols - 2; j++) {
            const sum = matrix[i][j] + matrix[i][j + 1] + matrix[i][j + 2] +
                        matrix[i + 1][j + 1] +
                        matrix[i + 2][j] + matrix[i + 2][j + 1] + matrix[i + 2][j + 2];
            maxSum = Math.max(maxSum, sum);
        }
    }
    return maxSum;
}

const matrix6 = [
    [1, 1, 1, 0, 0, 0],
    [0, 1, 0, 0, 0, 0],
    [1, 1, 1, 0, 0, 0],
    [0, 0, 2, 4, 4, 0],
    [0, 0, 0, 2, 0, 0],
    [0, 1, 1, 2, 4, 0]
];
console.log('Maximum Hourglass Sum:', hourglassSum(matrix6)); 

File: src/hourglassSum.java
public class hourglassSum {
    public static void main(String[] args) {

        int[][] myArr = {
                {-9, -9, -9, 1, 1, 1},
                {0, -9, 0, 4, 3, 2},
                {-9, -9, -9, 1, 2, 3},
                {0, 0, 8, 6, 6, 0},
                {0, 0, 0, -2, 0, 0},
                {0, 0, 1, 2, 4, 0}
        };

        int maxHourglassSum = findMaxHourglassSum(myArr);
        System.out.println("The maximum hourglass sum is: " + maxHourglassSum);
    }

    public static int findMaxHourglassSum(int[][] myArr) {
        int maxSum = Integer.MIN_VALUE;

        for (int i = 0; i <= myArr.length - 3; i++) {
            for (int j = 0; j <= myArr[0].length - 3; j++) {
                 int currentSum = calculateHourglassSum(myArr, i, j);
                if (currentSum > maxSum) {
                    maxSum = currentSum;
                }
            }
        }
        return maxSum;
}

public static int calculateHourglassSum(int[][] myArr, int startRow, int startCol) {
    int sum = 0;

    sum += myArr[startRow][startCol];
    sum += myArr[startRow][startCol + 1];
    sum += myArr[startRow][startCol + 2];

    sum += myArr[startRow + 1][startCol + 1];

    sum += myArr[startRow + 2][startCol];
    sum += myArr[startRow + 2][startCol + 1];
    sum += myArr[startRow + 2][startCol + 2];

    return sum;
}
}